src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java

```diff
@@ -22,70 +21,35 @@
 import org.apache.lucene.index.SortedDocValues;
 import org.apache.lucene.index.SortedNumericDocValues;
 import org.apache.lucene.index.SortedSetDocValues;
-import org.apache.lucene.store.Directory;
-import org.apache.lucene.store.FSDirectory;
-import org.apache.lucene.store.FilterDirectory;
-import org.opensearch.common.io.PathUtils;
-import org.opensearch.knn.common.FieldInfoExtractor;
 import org.opensearch.knn.index.codec.util.KNNCodecUtil;
-import org.opensearch.knn.index.engine.KNNEngine;
 import org.opensearch.knn.index.memory.NativeMemoryCacheManager;
 
 import java.io.IOException;
-import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-import static org.opensearch.knn.common.KNNConstants.MODEL_ID;
-import static org.opensearch.knn.index.mapper.KNNVectorFieldMapper.KNN_FIELD;
-
 @Log4j2
 public class KNN80DocValuesProducer extends DocValuesProducer {
-
-    private final SegmentReadState state;
     private final DocValuesProducer delegate;
-    private final NativeMemoryCacheManager nativeMemoryCacheManager;
-    private final Map<String, String> indexPathMap = new HashMap();
+    private final Map<String, String> fieldNameToVectorFileName = new HashMap<>();
 
     public KNN80DocValuesProducer(DocValuesProducer delegate, SegmentReadState state) {
         this.delegate = delegate;
-        this.state = state;
-        this.nativeMemoryCacheManager = NativeMemoryCacheManager.getInstance();
-
-        Directory directory = state.directory;
-        // directory would be CompoundDirectory, we need get directory firstly and then unwrap
-        if (state.directory instanceof KNN80CompoundDirectory) {
-            directory = ((KNN80CompoundDirectory) state.directory).getDir();
-        }
 
-        Directory dir = FilterDirectory.unwrap(directory);
-        if (!(dir instanceof FSDirectory)) {
-            log.warn("{} can not casting to FSDirectory", directory);
-            return;
-        }
-        String directoryPath = ((FSDirectory) dir).getDirectory().toString();
         for (FieldInfo field : state.fieldInfos) {
-            if (!field.attributes().containsKey(KNN_FIELD)) {
-                continue;
-            }
             // Only segments that contains BinaryDocValues and doesn't have vector values should be considered.
             // By default, we don't create BinaryDocValues for knn field anymore. However, users can set doc_values = true
             // to create binary doc values explicitly like any other field. Hence, we only want to include fields
             // where approximate search is possible only by BinaryDocValues.
-            if (field.getDocValuesType() != DocValuesType.BINARY || field.hasVectorValues() == true) {
+            if (field.hasVectorValues() || field.getDocValuesType() != DocValuesType.BINARY) {
```

Comment:
nit: why change this?

Reply:
Ah.. this is not sure... haha
Reverted it.